monai/handlers/ignite_metric.py

```diff
@@ -52,18 +59,37 @@ class IgniteMetric(Metric):
             https://github.com/Project-MONAI/tutorials/blob/master/modules/batch_output_transform.ipynb.
         save_details: whether to save metric computation details per image, for example: mean_dice of every image.
             default to True, will save to `engine.state.metric_details` dict with the metric name as key.
+        reduction: Argument for the LossMetric, look there for details
+        get_not_nans: Argument for the LossMetric, look there for details
 
     """
 
     def __init__(
-        self, metric_fn: CumulativeIterationMetric, output_transform: Callable = lambda x: x, save_details: bool = True
+        self,
+        metric_fn: CumulativeIterationMetric | None = None,
+        loss_fn: _Loss | None = None,
+        output_transform: Callable = lambda x: x,
+        save_details: bool = True,
+        reduction: MetricReduction | str = MetricReduction.MEAN,
+        get_not_nans: bool = False,
     ) -> None:
         self._is_reduced: bool = False
-        self.metric_fn = metric_fn
+        self.metric_fn: CumulativeIterationMetric = cast(CumulativeIterationMetric, metric_fn)
+        self.loss_fn = loss_fn
         self.save_details = save_details
         self._scores: list = []
         self._engine: Engine | None = None
         self._name: str | None = None
+
+        if self.metric_fn is None and self.loss_fn is None:
+            raise ValueError("Either metric_fn or loss_fn have to be passed.")
+        if self.metric_fn is not None and self.loss_fn is not None:
+            raise ValueError("Either metric_fn or loss_fn have to be passed, but not both.")
+        if self.loss_fn:
+            self.metric_fn = LossMetric(loss_fn=self.loss_fn, reduction=reduction, get_not_nans=get_not_nans)
+        # else:
```

Comment:
Seem we can remove these two comment lines?

Reply:
Yes, I forgot to remove those